docs/topics/exceptions.md

```diff
@@ -1,122 +1,705 @@
+This means `Nothing` can be used as a return type or generic type where any other type is expected, without causing type errors.
+
+`Nothing` is a special type in Kotlin used to represent functions or expressions that never complete successfully,
+either because they always throw an exception or enter an endless execution path like an infinite loop.
+You can use `Nothing` to mark functions that are not yet implemented or are designed to always throw an exception,
+clearly indicating your intentions to both the compiler and code readers.
+If the compiler infers a `Nothing` type in a function signature, it will warn you.
+Explicitly defining `Nothing` as the return type can eliminate this warning.
+
+This Kotlin code demonstrates the use of the `Nothing` type, where the compiler marks the code following the function
+call as unreachable:
+
+```kotlin
+class Person(val name: String?)
+
 fun fail(message: String): Nothing {
     throw IllegalArgumentException(message)
+    // This function will never return successfully.
+    // It will always throw an exception.
+}
+
+fun main() {
+    // Creates an instance of Person with 'name' as null
+    val person = Person(name = null)
+    
+    val s: String = person.name ?: fail("Name required")
+
+    // 's' is guaranteed to be initialized at this point
+    println(s)
 }
 ```
+{kotlin-runnable="true"}
 
-When you call this function, the compiler will know that the execution doesn't continue beyond the call:
+Kotlin's [TODO](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-t-o-d-o.html) function, which also uses the `Nothing` type, serves as a placeholder to highlight areas of the code that
+need future implementation:
 
 ```kotlin
-val s = person.name ?: fail("Name required")
-println(s)     // 's' is known to be initialized at this point
+fun notImplementedFunction(): Int {
+    TODO("This function is not yet implemented")
+}
+
+fun main() {
+    val result = notImplementedFunction()
+    // This throws a NotImplementedError
+    println(result)
+}
 ```
+{kotlin-runnable="true"}
 
-You may also encounter this type when dealing with type inference. The nullable variant of this type,
-`Nothing?`, has exactly one possible value, which is `null`. If you use `null` to initialize
-a value of an inferred type and there's no other information that can be used to determine a more
-specific type, the compiler will infer the `Nothing?` type:
+As you can see, the `TODO` function always throws a [NotImplementedError](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-not-implemented-error/).
+
+## Exception classes
+
+Let's explore some common exception types found in Kotlin, which are all subclasses of the [RuntimeException](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-runtime-exception/) class:
+
+* **[ArithmeticException](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-arithmetic-exception/)**: This exception occurs when an arithmetic operation is impossible to perform, like division by zero.
```

Comment:
**Bold** vs `Code`...
I prefer code here, but it is up to you

Reply:
Both look good to me (I don't have a preference), so I changed it to `code`